polaris-tokens/src/themes/light-uplift.ts

```diff
@@ -0,0 +1,182 @@
+    'color-border-interactive-hover': {value: colors.blue[14]},
+    'color-border-interactive': {value: colors.blue[13]},
+    'color-border-interactive-focus': {value: colors.blue[13]},
+    'color-border-interactive-subdued': {value: colors.blue[13]},
+    'color-border-magic-strong': {value: colors.purple[12]},
+    'color-border-magic': {value: colors.purple[10]},
+    'color-icon-hover': {value: colors.gray[15]()},
+    'color-icon': {value: colors.gray[14]()},
+    'color-icon-subdued': {value: colors.gray[12]()},
+    'color-icon-disabled': {value: colors.gray[10]()},
+    'color-icon-interactive-disabled': {value: colors.gray[10]()},
+    'color-icon-inverse': {value: colors.gray[8]()},
+    'color-icon-on-color': {value: colors.gray[1]()},
+    'color-icon-primary': {value: colors.gray[16]()},
+    'color-icon-success': {value: colors.green[12]},
+    'color-icon-critical': {value: colors.red[11]},
+    'color-icon-caution': {value: colors.yellow[11]},
+    'color-icon-info': {value: colors.azure[11]},
+    'color-icon-warning': {value: colors.orange[11]},
+    'color-icon-interactive-active': {value: colors.blue[15]},
+    'color-icon-interactive-hover': {value: colors.blue[14]},
+    'color-icon-interactive': {value: colors.blue[13]},
+    'color-icon-interactive-inverse': {value: colors.blue[8]},
+    'color-icon-magic': {value: colors.purple[13]},
+    'color-text': {value: colors.gray[15]()},
+    'color-text-subdued': {value: colors.gray[13]()},
+    'color-text-disabled': {value: colors.gray[11]()},
+    'color-text-interactive-disabled': {value: colors.gray[10]()},
+    'color-text-inverse-subdued': {value: colors.gray[10]()},
+    'color-text-inverse': {value: colors.gray[8]()},
+    'color-text-on-color': {value: colors.gray[1]()},
+    'color-text-success-strong': {value: colors.green[15]},
+    'color-text-success': {value: colors.green[15]},
+    'color-text-primary': {value: colors.gray[14]()},
+    'color-text-primary-hover': {value: colors.gray[14]()},
+    'color-text-critical-strong': {value: colors.red[14]},
+    'color-text-critical-active': {value: colors.red[16]},
+    'color-text-critical': {value: colors.red[14]},
+    'color-text-caution-strong': {value: colors.yellow[15]},
+    'color-text-caution': {value: colors.yellow[15]},
+    'color-text-info-strong': {value: colors.azure[16]},
+    'color-text-info': {value: colors.azure[14]},
+    'color-text-warning-strong': {value: colors.orange[16]},
+    'color-text-interactive-active': {value: colors.blue[15]},
+    'color-text-interactive-hover': {value: colors.blue[14]},
+    'color-text-interactive': {value: colors.blue[13]},
+    'color-text-interactive-inverse': {value: colors.blue[8]},
+    'color-text-magic-strong': {value: colors.purple[15]},
+    'color-text-magic': {value: colors.purple[14]},
+  },
+});
+
+export const themeLightUplift = deepmerge(
+  themeBase,
+  themeLightUpliftPartial,
+) as ThemeBase;
```

Comment:
Found it interesting that the variant theme gets cast here to the base theme. Not sure if we could make this clearer.

Reply:
Good call 👍 We can configure the `deepmerge` type parameters to properly intersect `ThemeBase` and `typeof  themeLightUplift` e.g.

```ts
export const themeLightUplift = deepmerge<ThemeBase, typeof themeLightUplift>(
  themeBase,
  themeLightUpliftPartial,
);
```

However, I find it bit verbose and the collective type utilities make it basically equivalent to the cast.

Perhaps, we can remove ambiguity with a `createThemeVariant` util. Thoughts?

```ts
export const themeLightUplift = createThemeVariant(themeLightUpliftPartial);
```

Reply:
Updated! https://github.com/Shopify/polaris/pull/10103/commits/24b86f27573a15de86100c838cbe59d7007dd6e2